lib/dependencies.sh

```diff
@@ -340,28 +340,32 @@ pnpm_prune_devdependencies() {
 
   cd "$build_dir" || return
 
-  # seems like we can't use prune without some workarounds because it triggers pre/post-install scripts
-  # and does not respect the `--ignore-scripts` argument:
-  # https://github.com/pnpm/pnpm/issues/5030
-  #
-  # so we're going to create a temporary .npmrc file which does allow us to set the ignore-scripts config:
-  # https://pnpm.io/npmrc#ignore-scripts
-  local app_npmrc="$build_dir/.npmrc"
-  local tmp_npmrc=$(mktemp)
-
-  # copy the original .npmrc if it exists
-  if [ -f "$app_npmrc" ]; then
-    cp "$app_npmrc" "$tmp_npmrc"
+  pnpm_version=$(pnpm --version)
+  pnpm_major_version=$(echo "$pnpm_version" | cut -d "." -f 1)
+  pnpm_minor_version=$(echo "$pnpm_version" | cut -d "." -f 2)
+  pnpm_patch_version=$(echo "$pnpm_version" | cut -d "." -f 3)
+
+  pnpm_prune_args=("prune" "--prod")
+
+  # prior to 8.15.6, pnpm prune would execute lifecycle scripts such as `preinstall` and `postinstall`
+  # so we should check if we're on that version + there are lifecycle scripts registered and, if so,
+  # we'll let the user know that pruning can't be done safely so we're skipping it
+  if (( "$pnpm_major_version" < 7 )) || \
+    (( "$pnpm_major_version" == 8 && "$pnpm_minor_version" < 15 )) || \
+    (( "$pnpm_major_version" == 8 && "$pnpm_minor_version" == 15 && "$pnpm_patch_version" < 6)); then
+      preinstall_script=$(read_json "$build_dir/package.json" ".scripts.preinstall")
+      postinstall_script=$(read_json "$build_dir/package.json" ".scripts.postinstall")
+      if [ -n "$preinstall_script" ] || [ -n "$postinstall_script" ]; then
```

Comment:
Wouldn't this also affect other scripts like `prepublish`, `preprepare`, `prepare`, and `postprepare`? You can see these all run with `npm ci` and `npm install`: https://docs.npmjs.com/cli/v10/using-npm/scripts. I'm guessing `pnpm` would mirror that behavior?

Reply:
Good catch. I was a bit too focused on the pre/post install scripts and forgot to check the complete list. Turns out `pnpm` does deviate slightly from the `npm` behavior here.  I expanded the test cases to include the full `npm` list [here](https://docs.npmjs.com/cli/v10/using-npm/scripts#npm-install) even though `pnpm` only runs the following subset (+1 additional script):
- `pnpm:devPreinstall`
- `preinstall`
- `install`
- `postinstall`
- `prepare`